+bids/File2.m

```diff
@@ -155,12 +165,25 @@
         end
       end
 
-      if ~isempty(obj.suffix)
+      obj.check_required_entities();
+
+      if isempty(obj.suffix)
```

Comment:
See my comment on the tests to see if when we want to throw a warning or error when there is no suffix.